sdk-utils/broadcast/broadcast.go

```diff
@@ -361,54 +355,59 @@ func (b *batchedBroadcaster) processBacklog() {
 		if b.backlog.Len() < b.batchThreshold {
 			task := b.backlog.Pop()
 
-			b.logger.Debug("low traffic; no batch merging", "batch_size", len(task.Msgs))
-			b.broadcast(task)
```

Comment:
why remove the `broadcast` function? the previous is cleaner to me

Reply:
the broadcast function before was actually "cheating". It created a completely new context that couldn't time out or be cancelled, so potentially it could block indefinitely. After properly deriving a context from the existing ones, the `broadcast` function didn't do anything but call `broadcaster.Broadcast` anymore, so I inlined the rest.